components/centraldashboard/test/e2e.test.ts

```diff
@@ -0,0 +1,144 @@
+import 'jasmine';
+
+import {Credentials, JWT} from 'google-auth-library';
+import {launch, Page} from 'puppeteer';
+import request from 'request';
+
+
+/**
+ * Look here to see setup instructions for Environment Setup for Cluster Auth:
+ *   https://cloud.google.com/kubernetes-engine/docs/tutorials/authenticating-to-cloud-platform
+ * 
+ * Use Scope: Owner
+ */ 
+const ORIGINAL_TIMEOUT = jasmine.DEFAULT_TIMEOUT_INTERVAL;
+const {
+    KF_HOST,   // The url to your cluster, usually: https://<cluster>.endpoints.<project>.cloud.goog/
+    CLIENT_ID, // This is the CLIENT_ID token used to create the cluster (used in http://deploy.kubeflow.cloud/)
+    SERVICE_ACCOUNT_EMAIL, // The email address for the service account you created <name>@<project>.iam.gserviceaccount.com
+    SERVICE_ACCOUNT_KEY,   // Should be a JSON file downloaded from cloud console service accounts
+} = process.env;
+
+['KF_HOST', 'CLIENT_ID', 'SERVICE_ACCOUNT_EMAIL', 'SERVICE_ACCOUNT_KEY'].forEach(envVar => {
+    if (!process.env[envVar]) {console.log(`${envVar} environment variable must be set`);process.exit(1);}
+});
+if (!/^https?:\/\/\S+/.test(KF_HOST)) {console.log('Invalid HOST url provided, must be like http*://*');process.exit(1);}
+const HOST = KF_HOST.replace(/\/$/, '');
+
+function getAuth(): Promise<Credentials> {
+    const client = new JWT({
+        keyFile: SERVICE_ACCOUNT_KEY,
+        additionalClaims: {target_audience: CLIENT_ID}
+    });
+    return client.authorize();
+}
+
+function waitFor(seconds: number): Promise<void> {
+    return new Promise((r) => setTimeout(r, seconds * 1000));
+}
+
+/**
+ * This function should help any dev down the line quickly inspect all
+ *  request headers as they arrive in the browser
+ * @param page Page to intercept headers for
+ */
+async function interceptHeadersForDebugging(page: Page) {
+    await page.setRequestInterception(true);
+    page.on("request", request => {
+        console.log("Request:", request.url(), request.headers());
+        request.continue();
+    });
+}
+
+/**
+ * This function should help any dev down the line quickly inspect all
```

Comment:
nit: Is this comment the same with the one above?

Reply:
Fixed, thanks for catching that!